File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        HashMap<Character, Integer> hm = new HashMap<>();
        int l = 0;
        int res = 0;

        for(int r = 0; r < s.length(); r++) {
            hm.merge(s.charAt(r), 1, Integer::sum);

            if(r - l + 1 - Collections.max(hm.values()) > k) {
                hm.put(s.charAt(l), hm.get(s.charAt(l)) - 1);
                l++;
            }
            res = Math.max(res, (r - l + 1));
        }
        return res;

    }
}

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = {}
        maxCount = 0
        l = 0
        res = 0

        for r in range(len(s)):
            count[s[r]] = 1 + count.get(s[r], 0)
            maxCount = max(maxCount, count[s[r]])

            while (r - l + 1) - maxCount > k:
                count[s[l]] -= 1
                l += 1
            res = max(res, r -l + 1)

        return res